cpp/include/raft/neighbors/nn_descent_types.hpp

```diff
@@ -56,6 +56,8 @@ struct index_params : ann::index_params {
   size_t max_iterations            = 20;      // Number of nn-descent iterations.
   float termination_threshold      = 0.0001;  // Termination threshold of nn-descent.
   bool return_distances            = false;   // return distances if true
+  size_t n_clusters                = 2;       // number of clusters if using batching
+  bool do_batch                    = false;
```

Comment:
I think a more simplified user experience would be to default n_clusters to 1 and then remove the `do_batch` option. This way, all a user needs to do to enable batching is increase the number of clusters.